src/Session.cpp

```diff
@@ -114,24 +136,24 @@ Session::Session(Session && s)
 
 Session& Session::operator=(Session && s)
 {
-    if(this != s)
+    if(this != &s)
     {
         clean();
         this->content = s.content;
         this->actionsLog = s.actionsLog;
         this->userMap = s.userMap;
         purgeSession(s);
     }
-    return this*;
+    return *this;
 }
 
-void Session::purgeSession(Session&& s)
+void Session::purgeSession(Session& s)
 {
      /* Since the vectors are held by value we have to give it an empty one,
         which will be quickly deleted anyway
     */
```

Comment:
So the destructor won't go through the pointers and delete them. The pointers that I stole

Reply:
OK so why not just nullptr? won't it work?

Reply:
Isn't replacing the stack with an empty object more efficient than iterating over the vector and replacing with nullptr one by one?

Reply:
I don't mean putting nullptr to all.
i mean doing ;s.actionsLog=nullptr`

Reply:
actionsLog is saved on the stack (by value), not as a pointer. so I can't do that

Reply:
and =null?
I don't know.. it just seems funny to create somthing to be deleted...

Reply:
and =null?
I don't know.. it just seems funny to create somthing to be deleted...

Reply:
=null doesn't work (NULL in cpp is defined as 0 and it can't do vector = 0)
Also it's not really creating something, since the stack memory is already reserved the constructor for vector would only make that memory empty again. It's the closest to delete we can do here